lib/src/rules/null_safety.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Do not opt out of null safety.';
+
+const _details = r'''
+
+**DON'T** use unsound null safety, by specifying a dart version lower than `2.12`.
+
+**BAD:**
+```dart
+// @dart=2.8
+a {
+}
+```
+
+**GOOD:**
+```dart
+b {
+}
+''';
+
+class NullSafety extends LintRule implements NodeLintRule {
+  NullSafety()
+      : super(
+            name: 'null_safety',
```

Comment:
Although it isn't strictly a requirement, consider naming this with a verb phrase, something like `enable_null_safety`. We generally try to name lints so that they describe either what _should_ be done or what _shouldn't_ be done.

Reply:
I see, it is now `enable_null_safety`. Another suggestion is `ensure_sound_null_safety`.